lib/config/dynamic/merger.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamic
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/pkg/registry"
+)
+
+type merger struct {
+}
+
+func NewMerger() Merger {
+	return &merger{}
+}
+
+// Merge merges two config objects together. For now only
+// whitelist registries are merged.
+func (m *merger) Merge(orig, other *config.VirtualContainerHostConfigSpec) (*config.VirtualContainerHostConfigSpec, error) {
+	otherWl := ParseRegistries(other.RegistryWhitelist)
+	origWl := ParseRegistries(orig.RegistryWhitelist)
+	wl, err := origWl.Merge(otherWl, &whitelistMerger{})
+	if err != nil {
+		return nil, err
+	}
+
+	orig.RegistryWhitelist = nil
```

Comment:
what does that mean if two white lists do not contain each other? is that meant to disable white list check?
with this logic, the merge eventually return empty white list without error, and then every time white list check will return true